www/grid_view/src/module/grid.controller.coffee

```diff
@@ -0,0 +1,166 @@
+        if not angular.isArray(@tags)
+            @tags = [@tags]
+
+        settings = bbSettingsService.getSettingsGroup('Grid')
+        @revisionLimit = settings.revisionLimit.value
+        @changeFetchLimit = settings.changeFetchLimit.value
+        @buildFetchLimit = settings.buildFetchLimit.value
+
+        @buildsets = @data.getBuildsets(
+            limit: @changeFetchLimit
+            order: '-bsid'
+            property: ['bsid', 'sourcestamps']
+        )
+        @changes = @data.getChanges(
+            limit: @changeFetchLimit
+            order: '-changeid'
+        )
+        @builders = @data.getBuilders(
+            property: ['builderid', 'name']
+        )
+        @buildrequests = @data.getBuildrequests(
+            limit: @buildFetchLimit
+            order: '-buildrequestid'
+            property: ['buildrequestid', 'buildsetid', 'builderid']
+        )
+        @builds = @data.getBuilds(
+            limit: @buildFetchLimit
+            order: '-buildrequestid'
+        )
+
+        @buildsets.onChange = @changes.onChange = @builders.onChange = \
+            @buildrequests.change = @builds.onChange = @onChange
+
+    dataReady: ->
+        for collection in [@buildsets, @changes, @builders, @buildrequests, @builds]
+            if not (collection.$resolved and collection.length > 0)
+                return false
+        return true
+
+    dataFetched: ->
+        for collection in [@buildsets, @changes, @builders, @buildrequests, @builds]
+            if not collection.$resolved
+                return false
+        return true
+
+    onChange: =>
+        if not @dataReady()
+            return
+
+        changesByBSID = {}
+        @$scope.changes = []
+        branches = {}
+
+        changesBySSID = {}
+        for c in @changes
+            changesBySSID[c.sourcestamp.ssid] = c
+
+        for bset in @buildsets
+            change = changesBySSID[_.last(bset.sourcestamps).ssid]
+            unless change?
+                continue
+            change.bsid = bset.bsid
+            change.branch ?= 'master'
+            branches[change.branch] = true
+
+            if @$scope.changes.length > @revisionLimit
+                continue
+            if @$stateParams.branch and change.branch != @$stateParams.branch
+                continue
+
+            @$scope.changes.push(change)
+            changesByBSID[bset.bsid] = change
+
+        @$scope.changes.sort((a, b) -> a.bsid - b.bsid)
+
+        @$scope.branches = (br for br of branches)
+
+        requestsByBSID = {}
+        for req in @buildrequests
+            (requestsByBSID[req.buildsetid] ?= []).push(req)
+        buildByReqID = {}
+        for build in @builds
+            buildByReqID[build.buildrequestid] = build
+
+        for builder in @builders
+            builder.builds = {}
+
+        buildersById = {}
+        # find builds for the selected changes and associate them to builders
+        for c in @$scope.changes
```

Comment:
I have spent an entire day trying to write and make tests work but I am currently blocked with this error:

```
Error: [$injector:unpr] Unknown provider: dataServiceProvider <- dataService
```

I have took example on the console view plugin and I don't see the difference...